### PR TITLE
Add post_delete signal to ensure Tasks are removed from opensearch

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**
### Description of change

This PR aims at ensuring a Task is removed from opensearch through a post_delete signal once it gets deleted from the database

### Checklist

* [x] Has this branch been rebased on top of the current `main` branch?

* [x] Is the CircleCI build passing?

### CODE REVIEW

File: datahub/search/task/signals.py [hunk 1]
```diff
@@ -14,4 +15,16 @@ def sync_task_to_opensearch(instance):
     )
 
 
-receivers = (SignalReceiver(post_save, DBTask, sync_task_to_opensearch),)
+def remove_task_from_opensearch(instance):
+    """Remove investor profile from es."""
```
Comment (chopkinsmade) [inline datahub/search/task/signals.py:19]: Should mention the task being removed not the investor profile

File: datahub/search/task/signals.py [hunk 2]
```diff
@@ -14,4 +15,16 @@ def sync_task_to_opensearch(instance):
...
+    transaction.on_commit(
+        lambda pk=instance.pk: delete_document(DBTask, pk),
+    )
+
+
+receivers = (
+    SignalReceiver(post_save, DBTask, sync_task_to_opensearch),
+    SignalReceiver(
+        post_delete, DBTask, remove_task_from_opensearch,
```
Comment (chopkinsmade) [inline datahub/search/task/signals.py:28]: I get this stack trace locally when I delete a task, is it working for you?
image

Comment (samuele-mattiuzzo) [reply datahub/search/task/signals.py:28]: How bizarre. I'm happy to have a look at it together tomorrow, the `Task` model does subclass `BaseSearchModel` 🤔

Comment (samuele-mattiuzzo) [reply datahub/search/task/signals.py:29]: I thought I was looking at the correct model @chopkinsmade -> https://github.com/uktrade/data-hub-api/pull/5111/commits/11f3011d1d45a54478b847ef94f65e990f5995cc should fix it

**Review by chopkinsmade — APPROVED** [review]

**Review by cgsunkel — APPROVED** [review]

**Pull request merged**